src/System.Runtime.Extensions/src/System/IO/Path.Windows.cs

```diff
@@ -120,12 +137,87 @@ private unsafe static string NormalizePath(string path, bool fullCheck, int maxP
         {
             Contract.Requires(path != null, "path can't be null");
 
+            // If the path is in extended syntax, we don't need to normalize, but we still do some basic validity checks
+            if (path.StartsWith(LongPathPrefix, StringComparison.Ordinal))
```

Comment:
Can all of this long-path-specific validity checking be moved into a separate helper method?